Nonfunctional `clang` on Mariner 1.0

The `clang` binary for Mariner 1.0 cannot resolve `stddef.h`. This does not occur in Mariner 2.0. A workaround is to use the `clang-devel` package to pull in the `clang`-provided `stddef.h`, but I am not sure if this is intended behavior. For comparison, the Mariner 2.0 specfile for `clang` does provide the `stddef.h` header by default through `clang-libs`[^0].

A minimal reproducible example of this behavior is at https://github.com/onalante-msft/mariner-clang, which attempts to compile the following:
```c
#include <stdio.h>

int main() {
    printf("HELLO\n");
}
```
This fails with the following error when built for Mariner 1.0 on a clean Ubuntu 18.04 installation:
```
time="2022-07-11T03:06:17Z" level=debug msg="+ clang main.c -o mariner-clang"
time="2022-07-11T03:06:17Z" level=debug msg="In file included from main.c:1:"
time="2022-07-11T03:06:17Z" level=debug msg="/usr/include/stdio.h:33:10: fatal error: 'stddef.h' file not found"
time="2022-07-11T03:06:17Z" level=debug msg="#include <stddef.h>"
time="2022-07-11T03:06:17Z" level=debug msg="         ^~~~~~~~~~"
time="2022-07-11T03:06:17Z" level=debug msg="1 error generated."
time="2022-07-11T03:06:17Z" level=debug
time="2022-07-11T03:06:17Z" level=debug msg="error: Bad exit status from /var/tmp/rpm-tmp.Rd6DTv (%build)"
time="2022-07-11T03:06:17Z" level=debug msg="    user tmp does not exist - using root"
time="2022-07-11T03:06:17Z" level=debug msg="    Bad exit status from /var/tmp/rpm-tmp.Rd6DTv (%build)"
```
Please let me know if there is any additional information I can provide to help debug this.

[^0]: https://github.com/microsoft/CBL-Mariner/blob/2.0.20220709-2.0/SPECS/clang/clang.spec#L55

--- a/SPECS/clang/clang.spec
+++ b/SPECS/clang/clang.spec
@@ -3,24 +3,24 @@ Name:           clang
 Version:        8.0.1
 Release:        5%{?dist}
 License:        NCSA
-URL:            https://clang.llvm.org
-Source0:        https://github.com/llvm/llvm-project/releases/download/llvmorg-%{version}/cfe-%{version}.src.tar.xz
-Group:          Development/Tools
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
+Group:          Development/Tools
+URL:            https://clang.llvm.org
+Source0:        https://github.com/llvm/llvm-project/releases/download/llvmorg-%{version}/cfe-%{version}.src.tar.xz
 BuildRequires:  cmake
+BuildRequires:  libxml2-devel
 BuildRequires:  llvm-devel = %{version}
 BuildRequires:  ncurses-devel
-BuildRequires:  zlib-devel
-BuildRequires:  libxml2-devel
 BuildRequires:  python2-devel
+BuildRequires:  zlib-devel
 Requires:       %{name}-libs = %{version}-%{release}
 Requires:       libstdc++-devel
-Requires:       ncurses
-Requires:       llvm
-Requires:       zlib
 Requires:       libxml2
+Requires:       llvm
+Requires:       ncurses
 Requires:       python2
+Requires:       zlib
 
 %description
 The goal of the Clang project is to create a new C based language front-end: C, C++, Objective C/C++, OpenCL C and others for the LLVM compiler. You can get and build the source today.
@@ -49,7 +49,7 @@ export CXXFLAGS="`echo " %{build_cxxflags} " | sed 's/ -g//'`"
 
 mkdir -p build
 cd build
-cmake -DCMAKE_INSTALL_PREFIX=/usr   \
+cmake -DCMAKE_INSTALL_PREFIX=%{_prefix}   \
       -DCMAKE_BUILD_TYPE=Release    \
       -DLLVM_ENABLE_RTTI=ON         \
       -Wno-dev ..
@@ -68,7 +68,7 @@ make DESTDIR=%{buildroot} install
 cd build
 make clang-check
 
-%clean
+%{clean}
 rm -rf %{buildroot}/*
 
 %files
@@ -91,21 +91,32 @@ rm -rf %{buildroot}/*
 %{_includedir}/*
 
 %changelog
-*   Tue Feb 09 2021 Henry Beberman <henry.beberman@microsoft.com> 8.0.1-4
--   Enable RTTI (runtime type information) so other packages can depend on it.
-*   Fri Jun 12 2020 Henry Beberman <henry.beberman@microsoft.com> 8.0.1-3
--   Temporarily disable generation of debug symbols.
-*   Sat May 09 2020 Nick Samson <nisamson@microsoft.com> - 8.0.1-2
--   Added %%license line automatically
-*   Tue Mar 17 2020 Henry Beberman <henry.beberman@microsoft.com> 8.0.1-1
--   Update to 8.0.1. Fix Source0 URL. License verified.
-*   Tue Sep 03 2019 Mateusz Malisz <mamalisz@microsoft.com> 6.0.1-2
--   Initial CBL-Mariner import from Photon (license: Apache2).
-*   Thu Aug 09 2018 Srivatsa S. Bhat <srivatsa@csail.mit.edu> 6.0.1-1
--   Update to version 6.0.1 to get it to build with gcc 7.3
-*   Wed Jun 28 2017 Chang Lee <changlee@vmware.com> 4.0.0-2
--   Updated %check
-*   Fri Apr 7 2017 Alexey Makhalov <amakhalov@vmware.com> 4.0.0-1
--   Version update
-*   Wed Jan 11 2017 Xiaolin Li <xiaolinl@vmware.com>  3.9.1-1
--   Initial build.
+* Sun Jul 10 2022 onalante-msft <89409054+onalante-msft@users.noreply.github.com> - 8.0.1-5
+- Include runtime libraries in base package.
+
+* Tue Feb 09 2021 Henry Beberman <henry.beberman@microsoft.com> - 8.0.1-4
+- Enable RTTI (runtime type information) so other packages can depend on it.
+
+* Fri Jun 12 2020 Henry Beberman <henry.beberman@microsoft.com> - 8.0.1-3
+- Temporarily disable generation of debug symbols.
+
+* Sat May 09 2020 Nick Samson <nisamson@microsoft.com> - 8.0.1-2
+- Added %%license line automatically
+
+* Tue Mar 17 2020 Henry Beberman <henry.beberman@microsoft.com> - 8.0.1-1
+- Update to 8.0.1. Fix Source0 URL. License verified.
+
+* Tue Sep 03 2019 Mateusz Malisz <mamalisz@microsoft.com> - 6.0.1-2
+- Initial CBL-Mariner import from Photon (license: Apache2).
+
+* Thu Aug 09 2018 Srivatsa S. Bhat <srivatsa@csail.mit.edu> - 6.0.1-1
+- Update to version 6.0.1 to get it to build with gcc 7.3
+
+* Wed Jun 28 2017 Chang Lee <changlee@vmware.com> - 4.0.0-2
+- Updated %check
+
+* Fri Apr 7 2017 Alexey Makhalov <amakhalov@vmware.com> - 4.0.0-1
+- Version update
+
+* Wed Jan 11 2017 Xiaolin Li <xiaolinl@vmware.com> - 3.9.1-1
+- Initial build.

--- a/SPECS/clang/clang.spec
+++ b/SPECS/clang/clang.spec
@@ -68,18 +68,15 @@ make DESTDIR=%{buildroot} install
 cd build
 make clang-check
 
-%{clean}
-rm -rf %{buildroot}/*
-
 %files
 %defattr(-,root,root)
-%license LICENSE.TXT
 %{_bindir}/*
 %{_libexecdir}/*
 %{_datadir}/*
 
 %files libs
 %defattr(-,root,root)
+%license LICENSE.TXT
 %{_libdir}/clang/*
 %{_libdir}/*.so.*
 

--- a/SPECS/clang/clang.spec
+++ b/SPECS/clang/clang.spec
@@ -14,6 +14,7 @@ BuildRequires:  ncurses-devel
 BuildRequires:  zlib-devel
 BuildRequires:  libxml2-devel
 BuildRequires:  python2-devel
+Requires:       %{name}-libs = %{version}-%{release}
 Requires:       libstdc++-devel
 Requires:       ncurses
 Requires:       llvm
@@ -23,6 +24,12 @@ Requires:       python2
 
 %description
 The goal of the Clang project is to create a new C based language front-end: C, C++, Objective C/C++, OpenCL C and others for the LLVM compiler. You can get and build the source today.
+
+%package libs
+Summary:        Runtime library for clang
+
+%description libs
+Runtime library for clang.
 
 %package devel
 Summary:        Development headers for clang
@@ -69,15 +76,18 @@ rm -rf %{buildroot}/*
 %license LICENSE.TXT
 %{_bindir}/*
 %{_libexecdir}/*
-%{_libdir}/*.so.*
 %{_datadir}/*
+
+%files libs
+%defattr(-,root,root)
+%{_libdir}/clang/*
+%{_libdir}/*.so.*
 
 %files devel
 %defattr(-,root,root)
 %{_libdir}/*.so
 %{_libdir}/*.a
 %{_libdir}/cmake/*
-%{_libdir}/clang/*
 %{_includedir}/*
 
 %changelog

--- a/SPECS/clang/clang.spec
+++ b/SPECS/clang/clang.spec
@@ -1,7 +1,7 @@
 Summary:        C, C++, Objective C and Objective C++ front-end for the LLVM compiler.
 Name:           clang
 Version:        8.0.1
-Release:        4%{?dist}
+Release:        5%{?dist}
 License:        NCSA
 URL:            https://clang.llvm.org
 Source0:        https://github.com/llvm/llvm-project/releases/download/llvmorg-%{version}/cfe-%{version}.src.tar.xz

